src/renderer/components/dock/dock-tab.store.ts

```diff
@@ -1,25 +1,30 @@
 import { autorun, observable, reaction } from "mobx";
-import { autobind, createStorage } from "../../utils";
+import { autobind, createStorage, StorageHelper } from "../../utils";
 import { dockStore, TabId } from "./dock.store";
 
 interface Options<T = any> {
-  storageName?: string; // name to sync data with localStorage
-  storageSerializer?: (data: T) => Partial<T>; // allow to customize data before saving to localStorage
+  storageName?: string; // persistent key
+  storageSerializer?: (data: T) => Partial<T>; // allow to customize data before saving
```

Comment:
We aren't doing serialization here.

```suggestion
  storageFinalizer?: (data: T) => Partial<T>; // allow to customize data before saving
```

Reply:
Out of PR's scope

Reply:
I acknowledge that switching to `Set` and `Map` is out of scope, but why is this out of scope?

Reply:
To not postpone milestone, all these suggestions not-critical.

Reply:
Why is allowing incorrect names through in a PR postponing the milestone?

Reply:
I have nothing to say anymore. Too many why..

Reply:
That is not how PRs should work. This current name is highly misleading and we shouldn't not be restrict what we can talk about because a release is coming up. If something isn't good enough then we will just bump it to the next release.

Reply:
If the name is more appropriate it's an easy change, right? This function is not exported to extensions?

Reply:
This functionality moved out from options to `protected` class method (overwritten in `edit-resource.store.ts`)